### PR TITLE
Moved UART check over to upstream driver

### PR DESCRIPTION
**Pull request by HU90m**
Also added simple LED check

### CODE REVIEW

File: sw/cheri/tests/uart_check.cc [hunk 1]
```diff
@@ -68,19 +25,16 @@ extern "C" void rom_loader_entry(void *rwRoot)
...
 
   // Create a bounded capability to the UART
-  Capability<volatile OpenTitanUart> uart = root.cast<volatile OpenTitanUart>();
-  uart.address() = 0x80100000;
-  uart.bounds()  = 0x34;
+  Capability<volatile OpenTitanUart<>> uart = root.cast<volatile OpenTitanUart<>>();
+  uart.address() = UART_ADDRESS;
+  uart.bounds()  = UART_BOUNDS;
+
+  uart->init(BAUD_RATE);
```
Comment (marnovandermaas) [inline sw/cheri/tests/uart_check.cc:32]: How does the UART know the CPU frequency?

**Review by marnovandermaas — COMMENTED** [review]
I haven't tested this yet.

File: sw/cheri/tests/uart_check.cc (hunk 1 above)
Comment (HU90m) [reply sw/cheri/tests/uart_check.cc:32]: The `CPU_TIMER_HZ` is common among our bare metal software and the CHERIoT RTOS. *[Implementation here.](https://github.com/microsoft/cheriot-rtos/blob/e8dfab9d3955e4a536c4d334e1d94b891e08f2cb/sdk/include/platform/sunburst/platform-uart.hh#L79)*

Comment (marnovandermaas) [reply sw/cheri/tests/uart_check.cc:32]: Ok, so do we then still need it defined in: `sw/common/defs.h`?

Comment (HU90m) [reply sw/cheri/tests/uart_check.cc:32]: Yep we do

**Review by marnovandermaas — APPROVED** [review]
Ok, I tested these. The GPIO test works as expected. The UART test prints the "hello there" and echoes whatever I type. Great work!

**Pull request merged**